apps/managedidentity/managedidentity.go

```diff
@@ -75,13 +96,14 @@ func SystemAssigned() ID {
 var cacheManager *storage.Manager = storage.New(nil)
 
 type Client struct {
-	httpClient ops.HTTPClient
-	miType     ID
-	// source     Source reenable when required in future sources
+	httpClient          ops.HTTPClient
+	miType              ID
+	retryPolicyDisabled bool
```

Comment:
How about we revert the flag, smth like "internalRetryPolicyEnabled" which defaults to `true`

Reply:
You mean that its disabled by-default and you enable like 
```go
miSystemAssigned, err := mi.New(mi.SystemAssigned(), WithEnabledInternalRetryPolicy())
```